SADET.py

```diff
@@ -510,16 +524,21 @@ def main():
                                     + input_dir_cont_path + "/" + path_pattern + "\" (" + str(expected_matches) + " matches expected, " + str(extraction_matches) + " found).")
 
         # go through the LocalApp path patterns for sample-wise files
-        for file_pattern_type in ["sample_DNA", "sample_RNA", "sample_DNA_bcl", "sample_RNA_bcl"]:
+        for file_pattern_type in ["sample_DNA", "sample_DNA_bcl", "sample_DNA_SPD", "sample_DNA_SPE",
+                                  "sample_RNA", "sample_RNA_bcl", "sample_RNA_SPD", "sample_RNA_SPE"]:
             # if the LocalApp analysis was started from FASTQ files, skip looking for files generated from BCL input
             if ((file_pattern_type in ["sample_DNA_bcl", "sample_RNA_bcl"]) and (not from_BCL)):
                 continue
             # process DNA- and RNA- specific path patterns in turn
             XNA_sample_list = DNA_sample_list
-            if (file_pattern_type in ["sample_RNA", "sample_RNA_bcl"]):
+            if (file_pattern_type in ["sample_RNA", "sample_RNA_bcl", "sample_RNA_SPD", "sample_RNA_SPE"]):
```

Comment:
could this be simplified like so?

```suggestion
            if "RNA" in file_pattern_type:
```

Reply:
I have no super strong feelings about it, As you said, it works with the current set of categories but this might change in the future and then you might need to modify this. Defining the categories in their own variable might make this shorter but depending on where it is defined it will be more difficult to read the code. Maybe just leave it as is.

Reply:
If we go for the arrays, I suppose they should be defined close to where the categories are introduced and commented on. If we would like to refer to specific sets of groups using sub-string keywords, we could instead define rules for how the categories are being named, so as to prevent issues with sub-string referrals (e.g., names of path categories for files related to individual RNA samples should start with "RNA_sample", and no other category names should start with this prefix; the `IF`-test can then check whether given `file_pattern_type` starts with "RNA_sample"). If we can make this work, we wouldn't need to update the `IF` test when introducing a new RNA sample category, so that would be convenient.

Thinking about my "hypothetical" example from above (a sequencing run without RNA samples), I realize we actually should need new categories for such files (I suppose there might be no RNA analysis step logs if no RNA samples are specified in the sample sheet - I can test that). The same would likely apply to sequencing runs without DNA samples (to be fair, I don't know whether we have any runs of the latter type in InPreD.. but it's certainly possible to create those). This means splitting the "general_all" path category.

Reply:
I think your idea with the sample type (rna, dna) prefix sounds good and then we wouldn't need to update the condition.

I also don't see a sequencing run without dna but it is probably good practice to account for **any** case 👍